test/Tests/src/Semantic/Equals_Spec.enso

```diff
@@ -0,0 +1,150 @@
+from Standard.Base import all
+
```

Comment:
Can we add a test like:

```
type My_Type
    C1 x
    C2 a b

    sum self = case self of
        C1 x -> x
        C2 a b ->  a+b

    == self other = case other of
        _ : My_Type -> self.sum==other.sum
        _ -> False

...
My_Type.C1 23 . should_not_equal My_Type.C1 24
My_Type.C1 10 . should_equal My_Type.C2 7 3
```

IMO this test will be failing now due to how atom equality is written, but if I understand our discussions correctly, conceptually we want such a test to pass.

Reply:
I have added this test.